File: car-racing-tournament-api/Models/UserSeason.cs
using System.Text.Json.Serialization;

namespace car_racing_tournament_api.Models
{
    public class UserSeason
    {
        public Guid Id { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public virtual User User { get; set; } = default!;
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public virtual Season Season { get; set; } = default!;
        public Guid UserId { get; set; }
        public Guid SeasonId { get; set; }
        public UserSeasonPermission Permission { get; set; }
    }

    public enum UserSeasonPermission
    {
        Moderator,
        Admin
    }
}

File: car-racing-tournament-web/src/app/models/user-season.ts
import { Season } from "./season";
import { User } from "./user";
import { UserSeasonPermission } from "./user-season-permission";

export class UserSeason {
  constructor(
    public id: string,
    public user: User,
    public season: Season,
    public userSeasonPermission: UserSeasonPermission
  ) {}
}
